### ISSUE TITLE
New Date Picker

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#5645]
## ✨ Feature

We've had various issues with the existing strategy of using `input type=date` for date inputs.  This is a placeholder issue to update this with a new date picker.

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🎨 Design File

Include a link to the design file (if it exists).

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Criteria 1
- [ ] Criteria 2

**Comment by substrae** [comment]
Added some notes above. @RM-1978 @esizer can y'all have a think about this approach from an accessibility perspective? I'm convinced that rather than messing around with libraries, having three standard, accessible inputs is going to provide the best experience across browsers, ATs, and devices.

**Comment by esizer** [comment]
> Added some notes above. @RM-1978 @esizer can y'all have a think about this approach from an accessibility perspective? I'm convinced that rather than messing around with libraries, having three standard, accessible inputs is going to provide the best experience across browsers, ATs, and devices.

Makes sense to me! We will need to be careful on validation here since, while unlikely, it could allow users to enter an invalid date. Should we consider a min date and max date for validation as well? I'm thinking of things like:

 - Birthday should have max date of today
 - End date should have min date of start date
 - Start date should have max date of end date
 - etc.

**Comment by RM-1978** [comment]
> Thanks @substrae I agree with @esizer 100%.  I also think that adding min and max dates would provide good safeguards  against consequences for the user resulting from potential mistakes (e.g. unintended keystrokes).

**Comment by mnigh** [comment]
This would solve https://github.com/GCTC-NTGC/gc-digital-talent/issues/4314.

**Comment by brindasasi** [comment]
can we have this fable tested from storybook? @RM-1978

**Comment by RM-1978** [comment]
We have had this Fable Tested this component. See results in [Fable Date Picker Test results and requirements #6652](https://github.com/GCTC-NTGC/gc-digital-talent/discussions/6652) for a discussion of each of the user requirements coming out of this work with the Lab.
